### DISCUSSION TITLE
Customize build cache

### DISCUSSION THREAD

**Discussion by thomkrupa** [vercel/vercel#3498] [Ideas]
Hey,

Is there a way to add files to build cache? To optimize gatsby static build I want to create a plugin similar to [netlify-cache plugin](https://github.com/axe312ger/gatsby-plugin-netlify-cache/). Currently, our project's build takes up to 7-8 mins because we transform a lot of images. That could be optimized by putting the whole `public` output (or at least `public/static` and `public/page-data`) in the cache.

I know Gatsby's `.cache` is adding to the build cache (https://github.com/zeit/now/issues/3247) but it's not enough in our case.

**Comment by nkuehn** [comment]
Another use case for this (our use case):  

many gatsbyJS sites in a monorepository that are always built all at once (to test link integrity) but whose cache folders live inside the packages. Would for example require a way to tell what's cached with a glob `/websites/*/.cache`.  Or more generally pointing to site root locations in the repository tree like `/websites/*` and then leaving the rest to the general framework detection logic.

**Comment by lucleray** [comment]
@thomkrupa We're now saving the `public` directory to the cache for Gatsby deployments (see https://github.com/zeit/now/pull/3669).

@nkuehn The cached directories are relative to the [Root Directory](https://vercel.com/docs/v2/build-step#root-directory) that is set on the project. If the Root Directory of your Gatsby project is set to `/websites/my-website`, the `/websites/my-website/.cache` and `/websites/my-website/public` will be cached.

**Reply by lukechu10** [reply]
Hey there, is there any plan to allow arbitrary folders to be added to the build cache? I am using [`trunk`](https://trunkrs.dev) but the prebuilt binaries do not work due to a missing glibc. I can only use `trunk` via `cargo install trunk` which can take up to 10 minutes. If the `.cargo/bin` folder were cached, `trunk` would only need to be build once.

Thanks

**Comment by nkuehn** [comment]
@lucleray  thanks for the info!   Does not help me though because it's many website roots under /websites that need to be built as one consistent set.

**Comment by deep-gaurav** [comment]
Hi, I'm building a site made in rust, and it's compiled to wasm which takes quite long each time without cache. It would be helpful if it was possible to cache rust's target directory.

**Comment by quantizor** [comment]
I would also like this to be able to cache the pnpm store, it doesn't seem to be included in the caching targets like npm/yarn

**Comment by jesse-savary** [comment]
My org really needs this; our monorepo uses Yarn 2's [Zero Installs](https://yarnpkg.com/features/zero-installs) and Vercel is eating up our GitHub LFS bandwidth very quickly.
